### PR TITLE
ofs.uio:add peek dump and log level Command-line arguments

### PR DESCRIPTION
**Pull request by anandaravuri**
peek-dump Command-line arguments
--peek-dump address size
  Peek CSR start address size (e.g.--peek-dump 0x0 0x20).

log level  Command-line arguments
--log-level {verbose,info}
    log level to use

### CODE REVIEW

File: binaries/ofs.uio/uio/ofs_uio.py [hunk 1]
```diff
@@ -67,11 +68,13 @@
...
 UIO_PATTERN = re.compile("uio[0-9]+")
 
+LOG = logging.getLogger(None)
+
 
 def verify_pcie_address(pcie_address):
     m = BDF_PATTERN.match(pcie_address)
     if m is None:
-        print("Invalid pcie address format{}".format(pcie_address))
+        LOG.error("Invalid pcie address format{}".format(pcie_address))
```
Comment (tswhison) [inline binaries/ofs.uio/uio/ofs_uio.py:77]: When using the logging module, that docs say to prefer using the %s, %d, etc format specifiers and to append the arguments as parameters to the function call.
```python
  LOG.error("Invalid pcie address format %s", pcie_address)
```
https://docs.python.org/3/howto/logging.html#logging-variable-data

File: binaries/ofs.uio/uio/ofs_uio.py [hunk 2]
```diff
@@ -330,17 +334,17 @@ def verify_uio(self):
         uio_path = os.path.join("/sys/class/uio/", self.uio)
         if os.path.exists(uio_path):
             with open(os.path.join(uio_path, 'name'), 'r') as fd:
-                print("UIO Name:", fd.read().strip())
+                LOG.debug("UIO Name: %s", fd.read().strip())
         else:
-            print("Invalid input UIO:", self.uio)
+            LOG.error('Invalid input UIO:%s', self.uio)
```
Comment (tswhison) [inline binaries/ofs.uio/uio/ofs_uio.py:339]: 👍 These are ok.  Please fix the others that use .format()

File: binaries/ofs.uio/uio/ofs_uio.py [hunk 3]
```diff
@@ -96,19 +99,20 @@ def verify_json(file_json):
...
-            print('Decoding JSON has failed')
+            LOG.error('Decoding JSON has failed')
             raise argparse.ArgumentTypeError('Decoding JSON has failed', file_json)
 
         for i in data:
             if i.get('address') is not None:
                 if verify_hex(i.get('address')) is False:
                     raise argparse.ArgumentTypeError('{} file contains invalid hex address {} {}'
-                    .format(file_json, i.get('reg-name'), i.get('address')))
+                                                     .format(file_json, i.get('reg-name'),
```
Comment (tswhison) [inline binaries/ofs.uio/uio/ofs_uio.py:109]: Thanks for the updates.  I think there is one straggler here.

Comment (anandaravuri) [reply binaries/ofs.uio/uio/ofs_uio.py:109]: format used in ArgumentTypeError() function, not in LOG.debug() or LOG.info

Comment (tswhison) [reply binaries/ofs.uio/uio/ofs_uio.py:109]: I see, my mistake.

File: binaries/ofs.uio/uio/ofs_uio.py [hunk 4]
```diff
@@ -687,67 +714,71 @@ def main():
...
             value = uio.peek(args.region_index, args.bit_size, args.peek)
-            print('peek({}): {}'.format(hex(args.peek), hex(value)))
+            LOG.info('peek(0x%x): 0x%x', args.peek, value)
+
+        elif args.peek_dump is not None:
+            addr = int(args.peek_dump[0], 16)
+            for i in range(int(args.peek_dump[1], 16)):
+                value = uio.peek(args.region_index, args.bit_size, addr)
+                LOG.info('peek(0x%x): 0x%x', addr, value)
+                addr = addr + (int(args.bit_size/8))
```
Comment (tswhison) [inline binaries/ofs.uio/uio/ofs_uio.py:731]: Sorry, just noticed this one.
```suggestion
                addr = addr + (int(args.bit_size)/8)
```

**Review by tswhison — APPROVED** [review]

**Pull request merged**